pkg/cli/cli.go

```diff
@@ -910,3 +925,53 @@ func buildServiceRootEndpoint(serviceURL, zone string) string {
 	const version = "v1"
 	return client.BuildRootEndpoint(serviceURL, version, zone)
 }
+
+// Prints out state changes.
+//
+// Only use this printer to print user friendly state changes, this is not a logger.
+type statePrinter struct {
+	// Writer to print the messages to. Visual features won't be displayed if not linked to an interactive terminal.
+	Out io.Writer
+
+	// If true, visual features like colors and animations won't be displayed.
+	visualsOn bool
+}
+
+func newStatePrinter(out io.Writer, verbose bool) *statePrinter {
+	visualsOn := false
+	if f, ok := out.(*os.File); ok && term.IsTerminal(int(f.Fd())) && !verbose {
+		visualsOn = true
+	}
+	return &statePrinter{Out: out, visualsOn: visualsOn}
+}
+
+func (p *statePrinter) Print(msg string) {
+	p.print(statePrinterEntry{Message: msg})
+}
+
+func (p *statePrinter) PrintDone(msg string) {
+	p.print(statePrinterEntry{Message: msg, Done: true})
+}
+
+type statePrinterEntry struct {
+	// State description.
+	Message string
+	// True if state is completed.
+	Done bool
+}
```

Comment:
nit: this seems unnecessary now, just passing two parameters to print is fine

Reply:
done